// pdftojson/lib/parser.js
'use strict';

var _classCallCheck = require('babel-runtime/helpers/class-call-check')['default'];

var _createClass = require('babel-runtime/helpers/create-class')['default'];

var _interopRequireDefault = require('babel-runtime/helpers/interop-require-default')['default'];

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _cheerio = require('cheerio');

var _cheerio2 = _interopRequireDefault(_cheerio);

var Page = function Page(width, height) {
  var words = arguments[2] === undefined ? [] : arguments[2];

  _classCallCheck(this, Page);

  this.width = +width;
  this.height = +height;
  this.words = words;
};

var Text = (function () {
  function Text(xMin, xMax, yMin, yMax) {
    var text = arguments[4] === undefined ? '' : arguments[4];

    _classCallCheck(this, Text);

    this.xMin = +xMin;this.xMax = +xMax;
    this.yMin = +yMin;this.yMax = +yMax;
    this.text = text;
  }

  _createClass(Text, [{
    key: 'width',
    get: function get() {
      return this.xMax - this.xMin;
    }
  }, {
    key: 'height',
    get: function get() {
      return this.yMax - this.yMin;
    }
  }]);

  return Text;
})();

// Parse HTML strings into Word / Page object instances.
//
function parse(html, shouldSort) {
  var $ = _cheerio2['default'].load(html);

  return $('page').map(function (idx, pageElem) {
    var $page = $(pageElem);

    return new Page($page.attr('width'), $page.attr('height'), $page.children('word').map(function (idx, wordElem) {
      var $word = $(wordElem);

      return new Text($word.attr('xmin'), $word.attr('xmax'), $word.attr('ymin'), $word.attr('ymax'), $word.text().trim());
    }).get());
  }).get();
}

exports['default'] = { Page: Page, Text: Text, parse: parse };
module.exports = exports['default'];
